hastexo/public/js/main.js

```diff
@@ -307,6 +173,25 @@ function HastexoXBlock(runtime, element, configuration) {
             dialog.dialog(dialog_container);
         };
 
+        terminal_client.onidle = function(pause) {
+            /* We're idle; clear stack info.  */
+            stack = null;
+    
+            var dialog = $('#idle');
+            if (pause) {
+                dialog.find('.message').html('Your lab is currently active in a separate window');
```

Comment:
This could still use a UX improvement. :) Currently, this ends up saying:

> We think you're busy elsewhere.
> 
> Your lab is currently active in a separate window
>
> Click 'OK' to resume it.

Apart from the fact that we're missing a full-stop after "window", it's also not quite the right message. By saying "Click 'OK' to resume it." we suggest that the lab is suspended, which it clearly isn't.

Maybe we can fix this by saying "Click 'OK' to return to your terminal in this window.", which would fit the pop-out use-case and would *also* apply to resuming from suspend?

Reply:
Sure, done.